rocky/tools/forms/boefje.py

```diff
@@ -65,3 +63,66 @@ class BoefjeAddForm(BaseRockyForm):
         widget=forms.Select(choices=SCAN_LEVEL.choices),
         help_text=BOEFJE_SCAN_LEVEL_HELP_TEXT,
     )
+
+    def __init__(self, katalogus_client: KATalogusClientV1, plugin_id: str, created: str, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.katalogus_client = katalogus_client
+        self.plugin_id = plugin_id
+        self.created = created
+
+    def create_boefje_with_form_data(self, form_data, plugin_id: str, created: str):
+        arguments = [] if form_data["oci_arguments"] == "" else form_data["oci_arguments"].split()
+        consumes = [] if form_data["consumes"] == "" else form_data["consumes"].strip("[]").replace("'", "").split(", ")
+        produces = [] if form_data["produces"] == "" else form_data["produces"].split(",")
+        produces = [p.strip() for p in produces]
+        input_objects = []
+
+        for input_object in consumes:
+            input_objects.append(type_by_name(input_object))
+
+        return Boefje(
+            id=plugin_id,
+            name=form_data["name"],
+            created=created,
+            description=form_data["description"],
+            enabled=False,
+            type="boefje",
+            scan_level=form_data["scan_level"],
+            consumes=input_objects,
+            produces=produces,
+            schema=form_data["schema"],
+            oci_image=form_data["oci_image"],
+            oci_arguments=arguments,
+        )
+
+
+class BoefjeAddForm(BoefjeSetupForm):
+    def clean(self):
+        cleaned_data = super().clean()
+
+        plugin = self.create_boefje_with_form_data(cleaned_data, self.plugin_id, self.created)
+
+        try:
+            self.katalogus_client.create_plugin(plugin)
+        except ReadTimeout:
+            handle_existing_name(self, plugin.name)
+
+        return cleaned_data
+
+
+class BoefjeEditForm(BoefjeSetupForm):
+    def clean(self):
+        cleaned_data = super().clean()
+
+        plugin = self.create_boefje_with_form_data(cleaned_data, self.plugin_id, self.created)
+
+        try:
+            self.katalogus_client.edit_plugin(plugin)
+        except ReadTimeout:
```

Comment:
My only concern about this ReadTimeout is that is not specific for duplicate names, it is just a general thing and that the message might not fit what the ReadTimeout exactly was. Maybe we can fix it in another PR, or katalogus needs to throw a specific error here like `DuplicateNameError` for example.

Reply:
I agree. The katalogus_client should throw a DuplicateNameError (subclassed from a generic KatalogusError) The generic KatalogusError should be caught, next to the DuplicateNameError which we can catch more specifically.

Reply:
Should be fixed now. I've created a DuplicateNameError which will be handled by the form.